### PR TITLE
More Compare Methods: Cyan-Magenta and Blend

### LINKED ISSUE #650: cyan/magenta compare method

### LINKED ISSUE #651: blend compare method

### LINKED ISSUE #653: create a test for 2D image compare

### PR DESCRIPTION
**Pull request by PaulHax**
`cyan-magenta` method puts the fixed image on component 0, moving image on component 1
and changes the color map for fixed image to cyan, moving image to magenta.

`blend` method puts the fixed image on component 0, moving image on component 1
and changes the color maps for both to gray-scale.

For each of these methods there is a `imageMix` parameter.  `imageMix` changes the percent contribution the fixed vs moving image makes to the render by modifying the opacity transfer function. Value of 1 means max opacity for moving image, no opacity for fixed image.

closes #651
closes #650
closes #653

### CODE REVIEW

File: src/IO/Compare/Compare.cxx [hunk 1]
```diff
@@ -91,72 +91,142 @@ int Checkerboard(itk::wasm::Pipeline &pipeline, const TMovingImage *movingImage,
...
-    return EXIT_FAILURE;
-  }
+    // cyan-magenta or blend
+
+    // using FilterType = itk::ComposeImageFilter<FixedImageType>;
+    // auto filter = FilterType::New();
+    // filter->SetInput(0, fixedImage);
+    // filter->SetInput(1, rescaleFilter->GetOutput());
+
+    // using PipelineOutputType = typename itk::VectorImage<typename FixedImageType::PixelType, FixedImageType::ImageDimension>;
```
Comment (thewtex) [inline src/IO/Compare/Compare.cxx:178]: Perhaps we should also output a two-component VectorImage? With Checkboard, the second component is what `swapImageOrder` enables now? We could simplify to one pipeline, enable cyan/magenta with imageMix or grayscale blent with imageMix with or without checkerboard?

Comment (PaulHax) [reply src/IO/Compare/Compare.cxx:178]: Good idea!  Faster swap image order toggling.  

And you put your finger right on the last issue (in my mind) with this PR:  Only getting a 1 component image out in JS land with this ComposeImagerFilter...

File: doc/content/api/index.md [hunk 2]
```diff
@@ -255,8 +255,30 @@ Moving image must have been added last.
 
 ```
 { method: 'checkerboard', pattern: number[], swapImageOrder: boolean } ||
+{ method: 'cyan-magenta', imageMix: number, checkerboard: boolean } ||
```
Comment (thewtex) [inline doc/content/api/index.md:258]: Add `pattern`

File: doc/content/api/index.md [hunk 3]
```diff
@@ -255,8 +255,30 @@ Moving image must have been added last.
...
+{ method: 'blend', imageMix: number, checkerboard: boolean } ||
```
Comment (thewtex) [inline doc/content/api/index.md:259]: Add `pattern`

File: doc/content/api/index.md [hunk 4]
```diff
@@ -255,8 +255,30 @@ Moving image must have been added last.
...
 { method: 'disabled' }
 ```
 
+`method` can be `checkerboard`, `cyan-magenta` or `blend`. The moving image is
+re-sampled to the fixed image space.
+
+`checkerboard` picks pixels from the fixed and moving image to create a
```
Comment (thewtex) [inline doc/content/api/index.md:266]: Communicate that this is equivalent to `blend` with `imageMix` set to `0`

File: doc/content/api/index.md [hunk 5]
```diff
@@ -255,8 +255,30 @@ Moving image must have been added last.
...
+checkerboard pattern. The 2 components in the output image differ by the
+image order for each checkerboard box.
+
+`cyan-magenta` method puts the fixed image on component 0, moving image on component 1
+and changes the color map for fixed image to cyan, moving image to magenta.
+
+`blend` method puts the fixed image on component 0, moving image on component 1
+and changes the color maps for both to grayscale.
```
Comment (thewtex) [inline doc/content/api/index.md:274]: keep color map from the last selected image (don't change the color map)

File: doc/content/api/index.md [hunk 6]
```diff
@@ -255,8 +255,30 @@ Moving image must have been added last.
...
+
 `pattern` is an array with the number of checkerboard boxes for each dimension.
 If pattern === undefined, it defaults to 4 boxes across each dimension
+
+`imageMix` changes the percent contribution the fixed vs moving image makes to the
+render by modifying the opacity transfer function. Value of 1 means max opacity for
+moving image, 0 for fixed image.
+
+`swapImageOrder` changes the opacity transfer to just show the first or second
+component. Use this to toggle the checkerboard pattern.
```
Comment (thewtex) [inline doc/content/api/index.md:284]: Describe the relationship with `imageMix`

File: doc/content/docs/viewer.md [hunk 7]
```diff
@@ -48,3 +47,25 @@ An additional parameter can be added to force the slice viewing mode. Just add `
 
 ![ItkVtkViewer2D](./2dViewer.jpg)
 
+### Compare Images URL Parameters
```
Comment (thewtex) [inline doc/content/docs/viewer.md:50]: CC @tbirdsong

**Review by thewtex — APPROVED** [review]

**Pull request merged**